community/detectors/saltstack_cve_2021_25281/src/main/java/com/google/tsunami/plugins/detectors/saltstack/cve202125281/SaltStackCVE202125281VulnDetector.java

```diff
@@ -0,0 +1,143 @@
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/**
+ * A {@link VulnDetector} that detects the CVE-2021-25281 vulnerability.
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "SaltStackCVE202125281VulnDetector",
+    version = "1.0",
+    description = "This detector checks for SaltStack Salt-API Unauthenticated "
+        + "Remote Command Execution vulnerability (CVE-2021-25281).",
+    author = "C4o (syttcasd@gmail.com)",
+    bootstrapModule = SaltStackCVE202125281VulnDetectorBootstrapModule.class
+)
+public class SaltStackCVE202125281VulnDetector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+
+  @Inject
+  SaltStackCVE202125281VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri = NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + "run";
+    HttpHeaders httpHeaders = HttpHeaders.builder()
+        .addHeader(com.google.common.net.HttpHeaders.CONTENT_TYPE, MediaType.JSON_UTF_8.toString())
+        .build();
+    ByteString httpBody = ByteString.copyFromUtf8(
+        "{\"eauth\": \"auto\", \"client\": \"wheel_async\", "
+            + "\"fun\": \"pillar_roots.write\", \"data\": \"test\", \"path\": "
+            + "\"../../../../../../../../../../../../../../../../../../tmp/success\"}");
+    try {
+      HttpResponse response = httpClient.send(
+          post(targetUri).withEmptyHeaders().setHeaders(httpHeaders).setRequestBody(httpBody)
+              .build());
+      Optional<JsonElement> body = response.bodyJson();
+      if (response.status() == HttpStatus.OK && body.isPresent()) {
+        JsonElement jsonElement = body.get();
+        if (!jsonElement.isJsonObject()) {
+          return false;
+        }
+        if (!jsonElement.getAsJsonObject().has("return") ||
+            !jsonElement.getAsJsonObject().get("return").isJsonArray()) {
```

Comment:
[Google Java Code Style](https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break): When a line is broken at a non-assignment operator the break comes before the symbol, i.e.:

```
        if (!jsonElement.getAsJsonObject().has("return")
            || !jsonElement.getAsJsonObject().get("return").isJsonArray()) {
```